datacube_apps/stacker/fixer.py

```diff
@@ -0,0 +1,273 @@
+    # Typically for particular years: the first and last for each product
+    # But for 2016, nothing will be stacked
+    # So we could do it based on NetCDF properties, and for all single-dataset locations
+    # Tile -> locations -> unique locations -> tile
+    # Just regex the filenames
+
+    query = {kw: arg for kw, arg in kwargs.items() if kw in ['cell_index'] and arg is not None}
+    gw = datacube.api.GridWorkflow(index=index, product=config['product'].name)
+
+    time_query_list = task_app.year_splitter(*kwargs['time']) if 'time' in kwargs else [None]
+
+    for time_query in time_query_list:
+        cells = gw.list_cells(product=config['product'].name, time=time_query, **query)
+        for cell_index, cell in cells.items():
+            files_to_fix = get_single_dataset_paths(cell)
+            if files_to_fix:
+                for time, tile in cell.split('time'):
+                    source_path = tile.sources.values.item()[0].local_path
+                    if source_path in files_to_fix:
+                        tile = gw.update_tile_lineage(tile)
+                        start_time = '{0:%Y%m%d%H%M%S%f}'.format(pd.Timestamp(time).to_datetime())
+                        output_filename = get_filename(config, cell_index, start_time)
+                        _LOG.info('Fixing required for: time=%s, cell=%s. Output=%s',
+                                  start_time, cell_index, output_filename)
+                        yield dict(start_time=time,
+                                   tile=tile,
+                                   cell_index=cell_index,
+                                   output_filename=output_filename)
+
+
+def make_fixer_config(index, config, export_path=None, **query):
+    config['product'] = index.products.get_by_name(config['output_type'])
+
+    if export_path is not None:
+        config['location'] = export_path
+        config['index_datasets'] = False
+    else:
+        config['index_datasets'] = True
+
+    if not os.access(config['location'], os.W_OK):
+        _LOG.warning('Current user appears not have write access output location: %s', config['location'])
+
+    chunking = config['storage']['chunking']
+    chunking = [chunking[dim] for dim in config['storage']['dimension_order']]
+
+    var_param_keys = {'zlib', 'complevel', 'shuffle', 'fletcher32', 'contiguous', 'attrs'}
+    variable_params = {}
+    for mapping in config['measurements']:
+        varname = mapping['name']
+        variable_params[varname] = {k: v for k, v in mapping.items() if k in var_param_keys}
+        variable_params[varname]['chunksizes'] = chunking
+
+    config['variable_params'] = variable_params
+
+    config['taskfile_version'] = int(datacube.utils.datetime_to_seconds_since_1970(datetime.datetime.now()))
+
+    return config
+
+
+def get_history_attribute(config, task):
+    tile = task['tile']
+    input_path = str(tile.sources[0].item()[0].local_path)
+    # original_dataset = xr.open_dataset(input_path)
```

Comment:
The history field is one of the things that is incorrect and needs to be fixed.
I'll update the code to reflect this.